models/biocro/R/run.biocro.R

```diff
@@ -51,111 +44,177 @@ run.biocro <- function(lat, lon, metpath, soil.nc = NULL, config = config, coppi
-        indesn <- Position(function(x)x==dayn, WetDat[,"doy"], right = TRUE)
-        dayn <- indesn/24
+      if ("WS" %in% names(WetDat)) {
+        WetDat <- dplyr::rename(WetDat, windspeed = "WS")
       }
-    }
-    
-    HarvestedYield <- 0
-    if (genus == "Saccharum") {
-      tmp.result <- BioCro::caneGro(WetDat = WetDat, lat = lat, soilControl = l2n(config$pft$soilControl))
-      # Addin Rhizome an Grain to avoid error in subsequent script processing results
-      tmp.result$Rhizome <- 0
-      tmp.result$Grain <- 0
-    } else if (genus == "Salix") {
+
       if (i == 1) {
-        iplant <- config$pft$iPlantControl
-      } else {
-        iplant$iRhizome <- data.table::last(tmp.result$Rhizome)
-        iplant$iRoot <- data.table::last(tmp.result$Root)
-        iplant$iStem <- data.table::last(tmp.result$Stem)
-        
-        if ((i - 1)%%coppice.interval == 0) {
+        initial_values <- config$pft$initial_values
+      }
+
+      if (i > 1) { # TODO HarvestedYield is never used and coppice not applicable to all crops. Rethink?
+        if ((i - 1) %% coppice.interval == 0) {
           # coppice when remainder = 0
           HarvestedYield <- round(data.table::last(tmp.result$Stem) * 0.95, 2)
-        } else if ((i - 1)%%coppice.interval == 1) 
-        {
+        } else if ((i - 1) %% coppice.interval == 1) {
           # year after coppice
-          iplant$iStem <- iplant$iStem * 0.05
+          initial_values$Stem <- initial_values$Stem * 0.05
         }  # else { # do nothing if neither coppice year nor year following
       }
-      ## run willowGro
-      
-      tmp.result <- BioCro::willowGro(WetDat = WetDat,
-                              iRhizome = as.numeric(iplant$iRhizome), 
-                              iRoot = as.numeric(iplant$iRoot),
-                              iStem = as.numeric(iplant$iStem), 
-                              day1 = day1, dayn = dayn, 
-                              soilControl = l2n(config$pft$soilControl), 
-                              canopyControl = l2n(config$pft$canopyControl), 
-                              willowphenoControl = l2n(config$pft$phenoParms), 
-                              seneControl = l2n(config$pft$seneControl), 
-                              photoControl = l2n(config$pft$photoParms))
-      
-    } else if (genus == "Miscanthus") {
-      if (yeari == years[1]) {
-        iRhizome <- config$pft$iPlantControl$iRhizome
-      } else {
-        iRhizome <- data.table::last(tmp.result$Rhizome)
-        HarvestedYield <- round(data.table::last(tmp.result$Stem) * 0.95, 2)
+
+      tmp.result <- BioCro::Gro(
+        initial_values = initial_values,
+        parameters = config$pft$parameters,
+        varying_parameters = WetDat,
+        modules = config$pft$modules)
+
+      # save final state as initial values for next year
+      # TODO: Some pools should NOT start at 100% of previous season --
+      # need to account for harvest, decomposition, etc
```

Comment:
@justinmcgrath does the v1 model handle harvest, decomposition, senescence, etc.

Reply:
Harvest isn't handled, but could be handled in the same manner as coppicing. Tissues senesce and litter accumulates. There is no decomposition, but at this point if someone has a model, it would be easy to add.